src/CommonDCModelDropdown.php

```diff
@@ -212,14 +212,202 @@ public static function getSpecificValueToDisplay($field, $values, array $options
         return parent::getSpecificValueToDisplay($field, $values, $options);
     }
 
+    /**
+     * Get the itemtype for this model
+     *
+     * @return string
+     */
+    public function getItemtypeForModel(): string
+    {
+        return str_replace('Model', '', get_called_class());
+    }
+
+    /**
+     * Get the items in racks that are using this model
+     *
+     * @return array
+     */
+    public function getItemsRackForModel(): array
+    {
+        $itemtype = $this->getItemtypeForModel();
+        return array_filter(
+            (new Item_Rack())->find([
+                'itemtype' => $itemtype,
+                'items_id' => new QuerySubQuery([
+                    'SELECT' => 'id',
+                    'FROM'   => $itemtype::getTable(),
+                    'WHERE'  => [
+                        $this->getForeignKeyField() => $this->fields['id'],
+                    ]
+                ])
+            ]),
+            // Filter out items that are not using the model
+            fn ($item_rack) => $itemtype::getById($item_rack['items_id'])->fields[strtolower(get_called_class()) . 's_id'] == $this->fields['id']
+        );
+    }
+
+    /**
+     * Check if a cell is filled for a specific orientations, hpos and depth
+     *
+     * @param array $cell
+     * @param int $orientation front or rear
+     * @param int $hpos left, right or full
+     * @param float $depth
+     *
+     * @return bool
+     */
+    private function isCellFilled(array $cell, int $orientation, int $hpos, float $depth): bool
+    {
+        // If hpos is full, check if both left and right are filled
+        if ($hpos == Rack::POS_NONE) {
+            return $this->isCellFilled($cell, $orientation, Rack::POS_LEFT, $depth)
+                || $this->isCellFilled($cell, $orientation, Rack::POS_RIGHT, $depth);
+        }
+
+        if (isset($cell[$hpos])) {
+            // Get the first $depth * 4 units of the cell to check if they are filled
+            $accurateCell = array_slice(
+                $orientation ?
+                    array_reverse($cell[$hpos]) // If orientation is rear, reverse the array
+                    : $cell[$hpos],
+                0,
+                $depth * 4
```

Comment:
@ccailly 

Depth can be a float, did you test for this scenario ?

I don't remember why we do `*4` here but isn't there a risk of issue if depth is a float since array_slice only accept int values ?